diff -p -up resolvconf-1.41/etc/resolvconf/update.d/libc.symlink resolvconf-1.41/etc/resolvconf/update.d/libc
--- resolvconf-1.41/etc/resolvconf/update.d/libc.symlink	2008-08-05 15:13:34.000000000 +0200
+++ resolvconf-1.41/etc/resolvconf/update.d/libc	2008-08-05 15:21:17.000000000 +0200
@@ -20,7 +20,7 @@ PATH=/sbin:/bin:/usr/bin
 # /etc/resolv.conf is not a symbolic link
 
 # Default value
-REPORT_ABSENT_SYMLINK=y
+REPORT_ABSENT_SYMLINK=no
 TRUNCATE_NAMESERVER_LIST_AFTER_127=y
 
 # Default override
@@ -118,9 +118,15 @@ for N in $NMSRVRS ; do echo "nameserver 
 
 ### Put the file in place ###
 
+move_rslvcnf() {
+	cat "$TMPFILE" > "$DYNAMICRSLVCNFFILE"
+	rm -f "$TMPFILE"
+}
+
 # On init, just put it in place
 if [ "$1" = "-i" ] ; then
-	mv -f "$TMPFILE" "$DYNAMICRSLVCNFFILE"
+	ln -sf "${ETC}/resolv.conf" "$DYNAMICRSLVCNFFILE"
+	move_rslvcnf
 	exit 0
 fi
 
@@ -131,7 +137,7 @@ if [ -f "$DYNAMICRSLVCNFFILE" ] && [ "$(
 fi
 
 # The file has changed
-mv -f "$TMPFILE" "$DYNAMICRSLVCNFFILE"
+move_rslvcnf
 
 # Restart nscd
 NSCD_RESTARTED=""
diff -p -up resolvconf-1.41/man/resolvconf.8.symlink resolvconf-1.41/man/resolvconf.8
--- resolvconf-1.41/man/resolvconf.8.symlink	2008-03-24 19:55:26.000000000 +0100
+++ resolvconf-1.41/man/resolvconf.8	2008-08-05 15:13:34.000000000 +0200
@@ -120,12 +120,6 @@ When nameserver information is updated t
 /etc/resolvconf/update.d/libc writes a new resolver configuration
 file to /etc/resolvconf/run/resolv.conf and then runs the scripts in
 /etc/resolvconf/update-libc.d/.
-To make the resolver use the dynamically generated resolver configuration
-file the administrator should ensure that /etc/resolv.conf is a symbolic
-link to /etc/resolvconf/run/resolv.conf.
-This link is never modified by \fB/sbin/resolvconf\fR.
-If you find that \fI/etc/resolv.conf\fR is not being updated,
-check to see that the link is intact.
 .PP
 The GNU C Library resolver library isn't the only resolver library available.
 However, any resolver library that reads /etc/resolv.conf
