--- a/bin/resolvconf	2012-09-15 12:37:52.296501927 +0400
+++ b/bin/resolvconf	2012-09-10 13:55:45.000000000 +0400
@@ -7,7 +7,7 @@
 
 echo_usage() { echo "Usage: resolvconf (-d IFACE|-a IFACE|-u|--enable-updates|--disable-updates|--updates-are-enabled|--create-runtime-directories|--wipe-runtime-directories)" ; }
 
-PATH=/sbin:/bin
+PATH=/sbin:/bin:/usr/bin
 MYNAME="${0##*/}"
 # Note that /etc/resolvconf/run may be (and generally is) a symlink
 RUN_DIR=/etc/resolvconf/run
--- a/etc/resolvconf/update.d/libc	2012-09-15 12:38:35.067977283 +0400
+++ b/etc/resolvconf/update.d/libc	2012-09-10 13:55:45.000000000 +0400
@@ -16,7 +16,7 @@
 #
 
 set -e
-PATH=/sbin:/bin
+PATH=/sbin:/bin:/usr/bin
 
 [ -x /lib/resolvconf/list-records ] || exit 1
 
