--- resolvconf-1.41/etc/resolvconf/update.d/libc.orig	2009-08-18 18:21:47.000000000 +0100
+++ resolvconf-1.41/etc/resolvconf/update.d/libc	2009-08-18 18:26:28.000000000 +0100
@@ -149,12 +149,10 @@
 	&& [ -x /usr/sbin/nscd ] \
 	&& [ -x /etc/init.d/nscd ] \
 	&& [ -s /etc/nscd.conf ] \
-	&& grep -q '^[[:space:]]*enable-cache[[:space:]]\+hosts[[:space:]]\+yes' /etc/nscd.conf \
-	&& start-stop-daemon --stop --quiet --pidfile="$NSCD_PIDFILE" --exec /usr/sbin/nscd --retry=TERM/3/HUP/5 \
-	&& start-stop-daemon --stop --quiet --oknodo --exec /usr/sbin/nscd --retry=0/5
+	&& grep -q '^[[:space:]]*enable-cache[[:space:]]\+hosts[[:space:]]\+yes' /etc/nscd.conf
 then
-	echo "Stopped Name Service Cache Daemon: nscd."
-	/etc/init.d/nscd start && { sleep 0.1 ; NSCD_RESTARTED=yes ; }
+	echo "Restarting Name Service Cache Daemon: nscd (if needed)"
+	/etc/init.d/nscd condrestart && { sleep 0.1 ; NSCD_RESTARTED=yes ; }
 fi
 
 # Notify users of the resolver
