diff --git a/bin/list-records b/bin/list-records
index b1a9ef1..07c1b33 100755
--- a/bin/list-records
+++ b/bin/list-records
@@ -73,14 +73,22 @@ if [ -r /etc/resolvconf/interface-order ] ; then
 		' /etc/resolvconf/interface-order)"
 fi
 
+INTERFACES=
 shopt -s nullglob extglob
 # The nullglob option isn't essential since the test -s below will eliminate
 # any unexpanded patterns. The extglob option, however, is essential; it
 # makes the glob language of interface-order patterns as powerful as regexps.
 
+while read P; do
+    EXP=$(echo $P)
+    [ -n "$EXP" ] && INTERFACES="$INTERFACES $(/lib/resolvconf/list-by-metric $EXP)"
+done <<EOF
+$PATTERNS
+EOF
+
 # Pathname expansion occurs on the following line resulting, in general,
 # in multiple instances of filenames; duplicates must be removed.
-uniquify $PATTERNS
+uniquify $INTERFACES
 # Result is in RSLT
 
 for FLNM in $RSLT ; do
diff --git a/man/interface-order.5 b/man/interface-order.5
index b5e176a..87e6b97 100644
--- a/man/interface-order.5
+++ b/man/interface-order.5
@@ -10,6 +10,8 @@ resolvconf nameserver information records
 are processed
 by those resolvconf update scripts
 that consult this file.
+In OpenMandriva, the expansion of each line is sorted again
+by the metric value of the interfaces.
 (The name of the file is apt because
 a resolvconf nameserver information record
 is named after the interface with which it is associated.)
